src/keria/app/ipexing.py

```diff
@@ -266,3 +272,203 @@ def sendMultisigExn(agent, hab, ked, sigs, atc, rec):
+
+        # Now create the stream to send, need the signer seal
+        kever = hab.kever
+        seal = eventing.SealEvent(i=hab.pre, s="{:x}".format(kever.lastEst.s), d=kever.lastEst.d)
+
+        ims = eventing.messagize(serder=serder, sigers=sigers, seal=seal)
+        ims = ims + atc.encode("utf-8")
+
+        # make a copy and parse
+        agent.hby.psr.parseOne(ims=bytearray(ims))
+
+        agent.exchanges.append(dict(said=serder.said, pre=hab.pre, rec=rec, topic='credential'))
+        return agent.monitor.submit(serder.pre, longrunning.OpTypes.exchange, metadata=dict(said=serder.said))
+    
+class IpexAgreeCollectionEnd:
+    
+    @staticmethod
+    def on_post(req, rep, name):
+        """ IPEX Agree POST endpoint
+
+        Parameters:
+            req: falcon.Request HTTP request
+            rep: falcon.Response HTTP response
+            name (str): human readable name for AID
+
+        ---
+        summary: Reply to IPEX offer message acknowledged willingness to accept offered credential
+        description: Reply to IPEX offer message acknowledged willingness to accept offered credential
+        tags:
+           - Credentials
+        responses:
+           200:
+              description: long running operation of IPEX agree
+        
+        """
+        agent = req.context.agent
+        hab = agent.hby.habByName(name)
+        if hab is None:
+            raise falcon.HTTPNotFound(description=f"alias={name} is not a valid reference to an identifier")
+
+        body = req.get_media()
+
+        ked = httping.getRequiredParam(body, "exn")
+        sigs = httping.getRequiredParam(body, "sigs")
+        rec = httping.getRequiredParam(body, "rec")
+
+        route = ked['r']
+
+        match route:
+            case "/ipex/agree":
+                op = IpexAgreeCollectionEnd.sendAgree(agent, hab, ked, sigs, rec)
+            case _:
+                raise falcon.HTTPBadRequest(description=f"invalid route {route}")
+
+        rep.status = falcon.HTTP_200
+        rep.data = op.to_json().encode("utf-8")
+
+    @staticmethod
+    def sendAgree(agent, hab, ked, sigs, rec):
+        for recp in rec:  # Have to verify we already know all the recipients.
+            if recp not in agent.hby.kevers:
+                raise falcon.HTTPBadRequest(description=f"attempt to send to unknown AID={recp}")
+
+        # use that data to create th Serder and Sigers for the exn
+        serder = serdering.SerderKERI(sad=ked)
+        sigers = [coring.Siger(qb64=sig) for sig in sigs]
+
+        # Now create the stream to send, need the signer seal
+        kever = hab.kever
+        seal = eventing.SealEvent(i=hab.pre, s="{:x}".format(kever.lastEst.s), d=kever.lastEst.d)
+
+        ims = eventing.messagize(serder=serder, sigers=sigers, seal=seal)
+
+        # make a copy and parse
+        agent.hby.psr.parseOne(ims=bytearray(ims))
+
+        agent.exchanges.append(dict(said=serder.said, pre=hab.pre, rec=rec, topic='credential'))
```

Comment:
I'm unsure if we are supposed to do something similar to admit (`Admitter`) where we now store the ACDC attachment that was offered, or if just the message is OK.

Reply:
Just the message is OK here.